### PR TITLE
Add improvement to support to use Salesforce Sandbox accounts

### PR DESCRIPTION
**Pull request by sachinira**
## Purpose
$subject
Fixes: https://github.com/ballerina-platform/ballerina-extended-library/issues/451

## Automation tests
 - Unit tests 
   > Tested with the WSO2 Sandbox
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: asyncapi/salesforce/tests/Config.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+username = "sachi@wso2.com"
+password = "NiyamaKalaa@23014jEh5ApzDTj9VP91hJgnq08yH"
```
Comment (LakshanSS) [inline asyncapi/salesforce/tests/Config.toml:2]: Shall we remove this?

File: asyncapi/salesforce/data_types.bal [hunk 2]
```diff
@@ -33,8 +37,20 @@ public type ListenerConfig record {|
     string channelName;
     @display{label: "Replay ID", "description": "The replay ID to change the point in time when events are read"}
     int replayFrom = REPLAY_FROM_TIP;
+    @display{label: "Environment", "description": "The type of Salesforce environment"}
+    string environment = PRODUCTION;
```
Comment (LakshanSS) [inline asyncapi/salesforce/data_types.bal:41]: Isn't this a breaking change?

File: asyncapi/salesforce/Ballerina.toml [hunk 3]
```diff
@@ -4,7 +4,7 @@ keywords = ["Sales & CRM/Customer Relationship Management", "Cost/Freemium"]
 org = "ballerinax"
 name = "trigger.salesforce"
 repository = "https://github.com/ballerina-platform/asyncapi-triggers"
-version = "0.9.0"
+version = "0.9.1"
```
Comment (LakshanSS) [inline asyncapi/salesforce/Ballerina.toml:7]: Shall we bump the version to 0.10.0 as there are considerable number of changes are added

File: asyncapi/salesforce/data_types.bal (hunk 2 above)
Comment (sachinira) [reply asyncapi/salesforce/data_types.bal:41]: This is not breaking change as the existing codes are not affected by the change. As we don't expect PRODUCTION/DEVELOPER users to add the environment they use. This has already being an issue to the SANDBOX users. So can consider as a major change

File: asyncapi/salesforce/Ballerina.toml (hunk 3 above)
Comment (sachinira) [reply asyncapi/salesforce/Ballerina.toml]: Done

**Review by LakshanSS — APPROVED** [review]

**Pull request merged**